### PR TITLE
Add to Create Op, generateAddress

### PR DESCRIPTION
**Pull request by coffeepots**
This PR adds the `generateAddress` function and fills in contract address generation for the Create op code.

### CODE REVIEW

**Comment by mratsim** [comment]
Link #133

File: nimbus/utils/addresses.nim [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import nimcrypto, eth_common, rlp
+
+func generateAddress*(address: EthAddress, nonce: AccountNonce): EthAddress =
+  result[0..19] = keccak256.digest(rlp.encodeList(address, nonce).toOpenArray).data[12..31]
```
Comment (mratsim) [inline nimbus/utils/addresses.nim]: ah you can use `.data.toOpenArray(12, 31)` here

Comment (coffeepots) [reply nimbus/utils/addresses.nim]: Hmm I tried this but its longer and doesn't compile on my end:

    func generateAddress*(address: EthAddress, nonce: AccountNonce): EthAddress =
      result[0..19] = keccak256.digest(rlp.encodeList(address, nonce).toOpenArray).data.toOpenArray(12..31)

`addresses.nim(4, 84) Error: type mismatch: got <array[0..31, byte], HSlice[system.int, system.int]>`

**Comment by coffeepots** [comment]
On further investigation:

### Nonce definition:

> A scalar value equal to the number of transactions  sent  from  this  address  or, in  the  case of accounts with associated code, the **number of contract-creations** made by this account. Note we use one fewer than the sender’s nonce value; we assert that we have incremented the sender account’s nonce **prior** to this call, and so the **value used is the sender’s nonce at the beginning of the responsible transaction or VM operation**. The account’s nonce is initially defined as one

In other words, nonce incrementation preceeds what is used as the 'active' value.

    let currentNonce = db.getNonce(storageAddress)
    db.setNonce(storageAddress, currentNonce + 1)

However, we currently assume Nonce starts at zero: https://github.com/status-im/nimbus/blob/9e1be6438e724231da9b2079b8c951b79e28dbe1/nimbus/account.nim#L18

File: nimbus/utils/addresses.nim (hunk 1 above)
Comment (cheatfate) [reply nimbus/utils/addresses.nim]: @coffeepots but in your attempt `toOpenArray(12..31)` while @mratsim ask you to call `toOpenArray(12, 31)`.

Comment (coffeepots) [reply nimbus/utils/addresses.nim]: Ah I missread.

What is the advantage of using `toOpenArray` here? Hiding implementation details of `data`?

    func generateAddress*(address: EthAddress, nonce: AccountNonce): EthAddress =
      result[0..19] = keccak256.digest(rlp.encodeList(address, nonce).toOpenArray).data.toOpenArray(12, 31)

**Pull request merged**